go/tasks/plugins/array/core/state.go

```diff
@@ -212,7 +212,7 @@ func MapArrayStateToPluginPhase(_ context.Context, state *State, logLinks []*idl
 		if state.GetExecutionErr() != nil {
 			phaseInfo = core.PhaseInfoFailed(core.PhasePermanentFailure, state.GetExecutionErr(), nowTaskInfo)
 		} else {
-			phaseInfo = core.PhaseInfoFailure(ErrorK8sArrayGeneric, state.GetReason(), nowTaskInfo)
+			phaseInfo = core.PhaseInfoSystemRetryableFailure(ErrorK8sArrayGeneric, state.GetReason(), nowTaskInfo)
```

Comment:
@EngHabu I updated this based on your feedback :) Do we really want to retry these failures? Not sure what is being marked as PhasePermanentFailure

Reply:
I think you are right here, this should continue to be a regular failure.. sorry for the false alarm